include/glow/CodeGen/MemoryAllocator.h

```diff
@@ -119,35 +170,64 @@ class MemoryAllocator {
   /// Frees the allocation associated with \p handle.
   void deallocate(Handle handle);
 
-  /// \returns the high water mark for the allocated memory.
-  uint64_t getMaxMemoryUsage() const { return maxMemoryAllocated_; }
+  /// \returns the total size (in bytes) of the memory.
+  uint64_t getMemorySize() const { return memorySize_; }
 
-  /// \returns the size of the whole memory region that we can allocate segments
-  /// into.
-  uint64_t getMemorySize() const { return poolSize_; }
+  /// \returns the maximum memory usage (in bytes).
+  uint64_t getMaxMemoryUsage() const { return maxUsedSize_; }
 
   /// \returns the alignment boundary used to align segments.
   size_t getAlignment() const { return alignment_; }
 
+  /// \returns the allocation efficiency as a float between 0.0 and 1.0
+  /// which quantifies the efficiency of the allocation algorithm. An
+  /// efficiency value of 1.0 means the best theoretically possible. The
+  /// efficiency is not always 1.0 due to memory fragmentation.
+  float getAllocationEfficiency() const;
+
   /// \returns the name of the memory region.
   const std::string &getName() const { return name_; }
 
 private:
   /// The name of the memory region.
   std::string name_;
-  /// A list of live buffers.
-  std::list<Segment> allocations_;
-  /// The size of the memory region that we can allocate segments into.
-  uint64_t poolSize_;
-  /// This is the high water mark for the allocated memory.
-  uint64_t maxMemoryAllocated_{0};
+
+  /// A list of allocated segments.
+  std::list<Segment> segments_;
```

Comment:
I just renamed the variable from `allocations_` to `segments_`. The variable was `std::list` from the beginning because the functions `allocate`/`deallocate` introduce/erase segments dynamically.

Reply:
I see. What about other places where you use std::list? You seem to be using it very actively in this PR.

Reply:
I use list for containers which grow dynamically with `push_back` and it's not easy to know before hand it's size in order to reserve memory (for example the liveness information). The input allocations are also a list because of the same reason: the way it's constructed in `AllocationsInfo.cpp`.

Reply:
Yes, you don't know the exact size in advance, but you know the upper bound, e.g. the number of IR instructions, so you can just reserve it in the container from the beginning and it will be only one dynamic allocation. std::list created new dynamic allocations for each element you add. So, unless you need to add/remove elements in the middle, there is no good reason for lists.

But this is a relatively minor thing.

Reply:
The only place I use now std::list is:
- when I have list of live IDs in the algorithm which are consumed randomly (elements removed from anywhere)
- the front-end allocation list provided by the user.
Do you want still to change the allocation list to `std::vector`?